applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h

```diff
@@ -96,137 +68,131 @@ class TrilinosVariationalDistanceCalculationProcess
+                    "In 3D the element type is expected to be a tetrahedra" << std::endl;
             }
         }
 
-
         base_model_part.GetCommunicator().SumAll(NNode);
         base_model_part.GetCommunicator().SumAll(NElem);
 
-        if( base_model_part.GetCommunicator().MyPID() == 0 )
-        {
-            if( NNode == 0 ) KRATOS_THROW_ERROR(std::logic_error, "the model has no Nodes","");
-            if( NElem == 0 ) KRATOS_THROW_ERROR(std::logic_error, "the model has no Elements","");
+        if( base_model_part.GetCommunicator().MyPID() == 0 ){
+            KRATOS_ERROR_IF(NNode == 0) << "The model has no nodes" << std::endl;
+            KRATOS_ERROR_IF(NElem == 0) << "The model has no elements" << std::endl;
         }
 
-        //generate an auxilary model part and populate it by elements of type DistanceCalculationElementSimplex
+        // Generate an auxilary model part and populate it by elements of type DistanceCalculationElementSimplex
         this->ReGenerateDistanceModelPart(base_model_part);
 
-        //generate a linear strategy
-
+        // Generate a linear strategy
         // Scheme
-        typename BaseType::SchemeType::Pointer pscheme = typename BaseType::SchemeType::Pointer( new ResidualBasedIncrementalUpdateStaticScheme< TSparseSpace,TDenseSpace >() );
+        SchemePointerType pscheme = Kratos::make_shared<ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace, TDenseSpace> >();
 
         // Builder and Solver
-        int RowSizeGuess = (TDim == 2 ? 15 : 40);
-        typedef typename BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>::Pointer BuilderSolverTypePointer;
-        BuilderSolverTypePointer pBuilderSolver = BuilderSolverTypePointer(new TrilinosBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver >(mrComm,RowSizeGuess,plinear_solver) );
+        const int RowSizeGuess = (TDim == 2 ? 15 : 40);
+        BuilderSolverPointerType pBuilderSolver = Kratos::make_shared<TrilinosBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver > >(
+            mrComm, 
+            RowSizeGuess, 
+            plinear_solver);
 
         // Solution strategy
-        bool CalculateReactions = false;
-        bool ReformDofAtEachIteration = false;
-        bool CalculateNormDxFlag = false;
-
-        this->mp_solving_strategy = typename BaseType::SolvingStrategyType::Pointer( new ResidualBasedLinearStrategy<TSparseSpace,TDenseSpace,TLinearSolver>(*(this->mp_distance_model_part),pscheme,plinear_solver,pBuilderSolver,CalculateReactions,ReformDofAtEachIteration,CalculateNormDxFlag) );
+        const bool CalculateReactions = false;
+        const bool ReformDofAtEachIteration = false;
+        const bool CalculateNormDxFlag = false;
+
+        (this->mp_solving_strategy) = Kratos::make_shared<ResidualBasedLinearStrategy<TSparseSpace, TDenseSpace, TLinearSolver> >(
```

Comment:
I know you didn't come up with this but I think this could/should be a `unique_ptr`